pkg/lease/healthProbe.go

```diff
@@ -0,0 +1,49 @@
+package lease
+
+import (
+	"context"
+	"net"
+	"net/http"
+
+	"github.com/golang/glog"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+)
+
+// ServeHealthProbes starts a server to check healthz and readyz probes
+func ServeHealthProbes(stop <-chan struct{}, healthProbeBindAddress string, configCheck healthz.Checker) {
+	healthzHandler := &healthz.Handler{Checks: map[string]healthz.Checker{
+		"healthz-ping": healthz.Ping,
+		"configz-ping": configCheck,
+	}}
+	readyzHandler := &healthz.Handler{Checks: map[string]healthz.Checker{
+		"readyz-ping": healthz.Ping,
+	}}
+
+	mux := http.NewServeMux()
+	mux.Handle("/readyz", http.StripPrefix("/readyz", readyzHandler))
+	mux.Handle("/healthz", http.StripPrefix("/healthz", healthzHandler))
+
+	server := http.Server{
+		Handler: mux,
+	}
+
+	ln, err := net.Listen("tcp", healthProbeBindAddress)
+	if err != nil {
+		glog.Errorf("error listening on %s: %v", ":8000", err)
+		return
+	}
+
+	glog.Infof("health probes server is running...")
+	// Run server
+	go func() {
+		if err := server.Serve(ln); err != nil && err != http.ErrServerClosed {
+			glog.Fatal(err)
+		}
+	}()
+
+	// Shutdown the server when stop is closed
+	<-stop
```

Comment:
For the first two questions:

I found the way to create a HealthProbeServer is very similar with the https://github.com/open-cluster-management/multicloud-operators-foundation. 

In multicloud-operators-foundation, the first parameter is not `context.Done()`, it's actually a `ctx.Done()`and this ctx is from `ctx := signals.SetupSignalHandler()` : https://github.com/open-cluster-management/multicloud-operators-foundation/blob/92233355d0ed2b62d7408d5a5205dd621c0e2961/cmd/agent/agent.go#L77

The signal `var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}` will trigger this stop channal.

Reply:
And I didn't really get the third question. I will try to answer this based on my current understanding.

First, this doesn't stop the server here because the server is running in an independent goroutine. 

After we get a stop signal, it will continue on, and `server.Shutdown(context.Backgroud())` will be invoked.

Then the server will be shutdown gracefully.

Reply:
Yes, we followed the pattern in multicloud-operators-foundation for [#17342](https://github.com/open-cluster-management/backlog/issues/17342) as other addons were using operators. Thanks for the discussion and input on testing yesterday, @xuezhaojun

Reply:
@jlpadilla Will we need to shutdown the channel after it is running? Wondering if we need the stop <-chan struct{} in ServeHealthProbes.